incubator/virtualcluster/pkg/syncer/resources/storageclass/checker.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageclass
+
+import (
+	"fmt"
+	"sync"
+
+	v1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/virtualcluster/pkg/syncer/constants"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+)
+
+func (c *controller) StartPeriodChecker(stopCh <-chan struct{}) error {
+	if !cache.WaitForCacheSync(stopCh, c.storageclassSynced) {
+		return fmt.Errorf("failed to wait for caches to sync before starting Service checker")
+	}
+
+	wait.Until(c.checkStorageClass, c.periodCheckerPeriod, stopCh)
+	return nil
+}
+
+// checkStorageClass check if StorageClass keeps consistency between super master and tenant masters.
+func (c *controller) checkStorageClass() {
+	clusterNames := c.multiClusterStorageClassController.GetClusterNames()
+	if len(clusterNames) == 0 {
+		klog.Infof("tenant masters has no clusters, give up storage class period checker")
+		return
+	}
+
+	wg := sync.WaitGroup{}
+
+	for _, clusterName := range clusterNames {
+		wg.Add(1)
+		go func(clusterName string) {
+			defer wg.Done()
+			c.checkStorageClassOfTenantCluster(clusterName)
+		}(clusterName)
+	}
+	wg.Wait()
+
+	pStorageClassList, err := c.storageclassLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("error listing storageclass from super master informer cache: %v", err)
+		return
+	}
+
+	for _, pStorageClass := range pStorageClassList {
+		if pStorageClass.Labels[constants.PublicObjectKey] != "true" {
```

Comment:
can we reuse `publicStorageClass` this function?

Reply:
Done